### PR TITLE
Fix little issues

### PR DESCRIPTION
**Pull request by lionkor**

### CODE REVIEW

File: src/Common.cpp [hunk 1]
```diff
@@ -386,3 +387,51 @@ void SplitString(const std::string& str, const char delim, std::vector<std::stri
     }
 }
 
+std::vector<uint8_t> DeComp(std::span<const uint8_t> input) {
+    beammp_debugf("got {} bytes of input data", input.size());
+
+    std::vector<uint8_t> output_buffer(std::min<size_t>(input.size() * 5, 15 * 1024 * 1024));
```
Comment (jimkoen) [inline src/Common.cpp]: What about a short comment detailing the constants in the vector constructor? As someone who hasn't touched this code it isn't immediately apparent what these do.

File: src/Common.cpp [hunk 2]
```diff
@@ -386,3 +387,51 @@ void SplitString(const std::string& str, const char delim, std::vector<std::stri
...
+    uLongf output_size = output_buffer.size();
+
+    while (true) {
+        int res = uncompress(
+            reinterpret_cast<Bytef*>(output_buffer.data()),
+            &output_size,
+            reinterpret_cast<const Bytef*>(input.data()),
+            static_cast<uLongf>(input.size()));
+        if (res == Z_BUF_ERROR) {
+            if (output_buffer.size() > 30 * 1024 * 1024) {
```
Comment (jimkoen) [inline src/Common.cpp]: Or just a short description why this is needed.

**Review by jimkoen — COMMENTED** [review]
Wasn't so sure about the magic constants, but I need to clarify that I never touch this code.

Other than that, lgtm

**Review by jimkoen — APPROVED** [review]

File: src/Common.cpp (hunk 1 above)
Comment (lionkor) [reply src/Common.cpp]: good point, thanky ou

File: src/Common.cpp (hunk 2 above)
Comment (lionkor) [reply src/Common.cpp]: good point here as well, thanks

File: src/Common.cpp [hunk 3]
```diff
@@ -386,3 +387,63 @@ void SplitString(const std::string& str, const char delim, std::vector<std::stri
...
+            beammp_warnf("zlib uncompress() failed, trying with a larger buffer size of {}", output_buffer.size());
+            output_size = output_buffer.size();
+        } else if (res != Z_OK) {
+            beammp_error("zlib uncompress() failed: " + std::to_string(res));
+            throw std::runtime_error("zlib uncompress() failed");
+        } else if (res == Z_OK) {
+            break;
+        }
+    }
+    output_buffer.resize(output_size);
```
Comment (jimkoen) [inline src/Common.cpp:429]: Isn't this superfluous given line 421?

**Review by jimkoen — DISMISSED** [review]
lgtm

**Pull request merged**